src/integration-tests/integration-test.py

```diff
@@ -478,6 +507,45 @@ def stop_template(self, target: str, delete_containers: bool = True) -> None:
                 stop_notifications=True,
             )
 
+    def check_log_end_marker(
+        self,
+    ) -> Tuple[bool, str, bool]:
+        logs = self.of.debug.logs.keyword(
+            self.stop_log_marker, limit=1, timespan="PT1H"
+        )
+        return (
+            len(logs["PrimaryResult"]) > 0,
+            "waiting for application insight logs to flush",
+            True,
+        )
+
+    def check_logs_for_errors(self) -> bool:
+        # only check for errors that exist between the start and stop markers
```

Comment:
It looks like the query has an additional condition: within the last 2 hours. Maybe worth emphasizing here, too?